openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/FeedRepository.java

```diff
@@ -752,18 +752,22 @@ public ResultList<Thread> list(
     String beforeCursor = null;
     String afterCursor = null;
     if (filter.getPaginationType() == PaginationType.BEFORE) {
-      if (threads.size()
-          > limit) { // If extra result exists, then previous page exists - return before cursor
-        threads.remove(0);
-        beforeCursor = threads.get(0).getUpdatedAt().toString();
+      if (!threads.isEmpty()) {
```

Comment:
add !nullOrEmpty here since threads is not initialized to - new ArrayList<>(); above

Reply:
also we can then move !nullOrEmpty outside of all ifs, since if that is null there is nothing to be done